crates/accelerate/src/unitary_synthesis.rs

```diff
@@ -0,0 +1,1094 @@
+            Some("rzx") => Some(StandardGate::RZXGate),     // 42
+            Some("xx_minus_yy") => Some(StandardGate::XXMinusYYGate), // 43
+            Some("xx_plus_yy") => Some(StandardGate::XXPlusYYGate), // 44
+            _ => None,
+        }
+    }
+}
+
+fn dag_from_2q_gate_sequence(
+    py: Python<'_>,
+    sequence: TwoQubitUnitarySequence,
+) -> PyResult<DAGCircuit> {
+    let gate_vec = &sequence.gate_sequence.gates;
+    let mut target_dag = DAGCircuit::new(py)?;
+    let _ = target_dag.set_global_phase(Param::Float(sequence.gate_sequence.global_phase));
+
+    let mut instructions = Vec::new();
+
+    let qubit: &Bound<PyAny> = imports::QUBIT.get_bound(py);
+    let mut qubits: Vec<Qubit> = vec![];
+
+    let qubit_obj = qubit.call0()?;
+    qubits.push(target_dag.add_qubit_unchecked(py, &qubit_obj)?);
+
+    for (gate, params, qubit_ids) in gate_vec {
+        let gate_node = match gate {
+            None => sequence.get_decomp_gate().unwrap(),
+            Some(gate) => *gate,
+        };
+
+        let mut gate_qubits = Vec::new();
+
+        for id in qubit_ids {
+            while *id as usize >= qubits.len() {
+                let qubit_obj = qubit.call0()?;
+                qubits.push(target_dag.add_qubit_unchecked(py, &qubit_obj)?);
+            }
+            gate_qubits.push(qubits[*id as usize]);
+        }
+
+        let new_params: Option<Box<SmallVec<[Param; 3]>>> = match gate {
+            Some(_) => Some(Box::new(params.iter().map(|p| Param::Float(*p)).collect())),
+            None => sequence
+                .decomp_gate_params
+                .as_ref()
+                .map(|params| Box::new(params.clone())),
+        };
+
+        let pi = PackedInstruction {
+            op: PackedOperation::from_standard(gate_node),
+            qubits: target_dag.qargs_interner.insert(&gate_qubits),
+            clbits: target_dag.cargs_interner.get_default(),
+            params: new_params,
+            extra_attrs: ExtraInstructionAttributes::new(None, None, None, None),
+            #[cfg(feature = "cache_pygates")]
+            py_op: OnceCell::new(),
+        };
+        instructions.push(pi);
+    }
+
+    let _ = target_dag.extend(py, instructions.into_iter());
+
+    Ok(target_dag)
+}
+
+// This is the outer-most run function. It is meant to be called from Python inside `UnitarySynthesis.run()`
+// This loop iterates over the dag and calls `run_2q_unitary_synthesis`
+#[pyfunction]
+#[pyo3(name = "run_default_main_loop")]
+fn py_run_default_main_loop(
+    py: Python,
+    dag: &mut DAGCircuit,
+    qubit_indices: &Bound<'_, PyList>,
+    min_qubits: usize,
+    target: &Target,
+    approximation_degree: Option<f64>,
+    neighbor_table: Option<&NeighborTable>,
+    distance_matrix: Option<PyReadonlyArray2<f64>>,
+    natural_direction: Option<bool>,
+) -> PyResult<DAGCircuit> {
+    let circuit_to_dag = imports::CIRCUIT_TO_DAG.get_bound(py);
+    let dag_to_circuit = imports::DAG_TO_CIRCUIT.get_bound(py);
```

Comment:
I updated the code to use `circuit_to_dag` from the new converters crate in [a912248](https://github.com/Qiskit/qiskit/pull/13141/commits/a91224861f21971e815187975afe13b010108531). I am hesitant with `dag_to_circuit` because to use it I'd have to add a few extra conversion steps.